homeassistant/components/logbook/__init__.py

```diff
@@ -474,6 +476,22 @@ def _get_events(
 
     def yield_rows(query: Query) -> Generator[Row, None, None]:
         """Yield Events that are not filtered away."""
+
+        def _keep_row(row: Row, event_type: str) -> bool:
```

Comment:
I made this a closure so we don't access `hass.data` every iteration.

Reply:
That will probably make more sense when you see https://github.com/home-assistant/core/pull/71830 which is the final PR to get `dev` in sync with my integration branch.